x-pack/plugin/autoscaling/src/test/java/org/elasticsearch/xpack/autoscaling/storage/ReactiveStorageDeciderDecisionTests.java

```diff
@@ -381,28 +420,37 @@ private static void verifyScale(ClusterState state, long expectedDifference, Str
         );
         TestAutoscalingDeciderContext context = createContext(state, Set.of(DiscoveryNodeRole.DATA_HOT_NODE_ROLE));
         AutoscalingDeciderResult result = decider.scale(Settings.EMPTY, context);
+        ReactiveStorageDeciderService.ReactiveReason resultReason = (ReactiveStorageDeciderService.ReactiveReason) result.reason();
+
         if (context.currentCapacity != null) {
             assertThat(
                 result.requiredCapacity().total().storage().getBytes() - context.currentCapacity.total().storage().getBytes(),
                 equalTo(expectedDifference)
             );
-            assertThat(result.reason().summary(), equalTo(reason));
+            assertThat(resultReason.summary(), equalTo(reason));
+            assertThat(resultReason.unassignedShardIds(), equalTo(decider.allocationState(context).storagePreventsAllocation().shardIds()));
+            assertThat(resultReason.assignedShardIds(), equalTo(decider.allocationState(context).storagePreventsRemainOrMove().shardIds()));
         } else {
             assertThat(result.requiredCapacity(), is(nullValue()));
-            assertThat(result.reason().summary(), equalTo("current capacity not available"));
+            assertThat(resultReason.summary(), equalTo("current capacity not available"));
+            assertThat(resultReason.unassignedShardIds(), equalTo(Set.of()));
+            assertThat(resultReason.assignedShardIds(), equalTo(Set.of()));
         }
     }
 
-    private long numAllocatableSubjectShards() {
+    record AllocatableShards(long numOfShards, SortedSet<ShardId> shardIds) {}
+
+    private AllocatableShards allocatableShards() {
```

Comment:
I deliberately collected the data to a list, because I believed have seen some test failing when the amount was different from the size of unique list of shards. Let me take a closer look and just use a `SortedSet` if it's not the case.

Reply:
Resolving this, since the size of the set is different from the number of shards when there are multiple shard copies.